Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -1009,10 +1003,10 @@ extension TaskHandler: ChannelDuplexHandler {
 
     func channelInactive(context: ChannelHandlerContext) {
         switch self.state {
+        case .inactive, .sending, .sent, .redirected:
+            self.errorCaught(context: context, error: HTTPClientError.remoteConnectionClosed)
```

Comment:
hmm, `channelInactive` in `.inactive` (which should probably be named `idle` because the channel isn't "inactive") is really remote connection closed? Isn't that error "between requests" which is expected for a server to do?

Reply:
`.idle` + `channelInactive` means channel was closed _before_ we sent request head, right now we do not re-use the handler between requests (we remove and add it between requests)

Reply:
@artemredkin Ahhh! I see, so we put in this handler just before we send `.head` and then we get inactive. That makes sense, thanks!